Additional "Default Data Path" changes

ttpcodes opened the pull request:
This PR makes some additional changes to the implementation of the `Default Data Path Setting` that were discussed in longhorn/longhorn#582:
- The default value of the `Setting` is now `/var/lib/longhorn`.
- There is a new `boolean Setting` titled `create-default-disk` that allows users to specify whether or not they'd like to have the default `Disk` automatically created.
- The description of the `Default Data Path Setting` has been fixed to use the verb `storing` instead of `mounting`.
- On `Node` creation, the `longhorn-manager` will attempt to enter the host `namespace` to create the directory specified by `Default Data Path`. This will create the directory if it doesn't already exist, and do nothing if it does already exist. Of course, if this fails, the `longhorn-manager` will still `panic`.

yasker commented on types/setting.go:
```diff
@@ -117,7 +117,7 @@ var (
 		Required:    true,
 		ReadOnly:    false,
-		Default:     "/var/lib/rancher/longhorn/",
+		Default:     "/var/lib/longhorn/",
```
Don't change the path for now. There are other things associated with the path. We need to change them together.

ttpcodes replied:
Dropping the change.

yasker commented on datastore/longhorn.go:
```diff
@@ -867,6 +869,13 @@ func (s *DataStore) CreateDefaultNode(name string) (*longhorn.Node, error) {
 		}
 
+		// Attempt to create the specified Default Data Path on the disk, in case it doesn't exist.
+		nsPath := iscsi_util.GetHostNamespacePath(util.HostProcPath)
```
Use `NamespaceExecutor`? https://github.com/longhorn/longhorn-manager/blob/master/util/util.go#L453

ttpcodes replied:
Changing the code to make use of `NamespaceExecutor`.

ttpcodes commented:
- Dropped the path change from `/var/lib/rancher/longhorn` to `/var/lib/longhorn`, since this will probably be addressed in a later PR along with other path changes.
- Modified the call to `mkdir` to instead use `NamespaceExecutor`.
